airbyte-config/init/src/main/java/io/airbyte/config/init/SeedRepository.java

```diff
@@ -125,4 +116,26 @@ public static void main(String[] args) throws IOException {
     new SeedRepository().run(idName, inputPath, outputPath);
   }
 
+  public void run(final String idName, final Path input, final Path output) throws IOException {
+    final JsonNode seedList = Yamls.deserialize(IOs.readFile(input));
```

Comment:
My heuristic is the rule of three - and we have just hit it (Seed Repository, SourceDefinitionHandler and DestinationDefinitionHandler) so let's do it!

Reply:
We end up at the same conclusion here, but just wanted to share a thought. I use this heuristic you mentioned alot too. I would actually say it's maybe not a good fit for this case though. 

Essentially we have are converting from a poorly defined object to a well defined object and coming up with a protocol on how to do it. The key in my mind that the because there's a poorly defined thing on one side (some raw yaml that we don't have an official struct for), it's easy to make a mistake if the code is duplicated because the compiler can't really help us. So my intuition here is that you don't want to centralize this logic and not have any duplication, because it is logic that is easy to mess up. and if the duplicated code does go out of sync, it will be hard to detect.